### PR TITLE
BUG Prevent extensions from decorating $owner property with redundant subclasses

### PR DESCRIPTION
**Pull request by tractorcow**
If Page has an extension HeaderExtension, then the HeaderExtension.php does not need to include ALL of the Page subclasess in the $owner property; The base class is enough. :)

### CODE REVIEW

File: src/Generators/AbstractTagGenerator.php [hunk 1]
```diff
@@ -176,21 +178,32 @@ public function getExistingTags()
...
-                $config = Config::inst()->get($class, 'extensions');
-
-                return ($config !== null && in_array($className, $config, null));
+                $config = Config::inst()->get(
+                    $class,
+                    'extensions',
+                    Config::UNINHERITED | Config::EXCLUDE_EXTRA_SOURCES
+                );
+                if (!$config) {
+                    return false;
```
Comment (Firesphere) [inline src/Generators/AbstractTagGenerator.php:199]: Return false?? The tests show it's correct, but it looks a bit weird to me.

File: src/Generators/AbstractTagGenerator.php [hunk 2]
```diff
@@ -176,21 +178,32 @@ public function getExistingTags()
...
+                }
+                foreach ($config as $candidateClass) {
+                    if (Extension::get_classname_without_arguments($candidateClass) === $className) {
+                        return true;
```
Comment (Firesphere) [inline src/Generators/AbstractTagGenerator.php:203]: Maybe, returning a return variable makes more sense? Also, this loop is unreliable. I feel a variable would make more sense, to keep the code semantically readable.

File: src/Generators/AbstractTagGenerator.php (hunk 1 above)
Comment (tractorcow) [reply src/Generators/AbstractTagGenerator.php:199]: That's a class with an empty extensions config :) E.g. `private static $extensions = [];`

File: src/Generators/AbstractTagGenerator.php (hunk 2 above)
Comment (tractorcow) [reply src/Generators/AbstractTagGenerator.php:203]: I think that makes sense actually. Updating.

Comment (tractorcow) [reply src/Generators/AbstractTagGenerator.php:203]: Or I'll make it a generator instead of array_map.

**Comment by tractorcow** [comment]
@Firesphere updated :)

**Comment by tractorcow** [comment]
I had to flick framework dependencies to stable tags for tests (I guess deleting the old branches broke the installs).

**Comment by tractorcow** [comment]
Tests green :)

**Review by robbieaverill — APPROVED** [review]
LGTM, @Firesphere's feedback has been addressed. Thanks for fixing the tests too.

**Pull request merged**